src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA = ",";
+    private static final int INNER_ARRAY_SIZE = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(toFileName, calculateReport(readFromFile(fromFileName)));
+    }
+
+    private String[][] readFromFile(String fileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            int i = 0;
+            int arraySize = createArraySize(fileName);
+            String line;
+            String[][] sourceArray = new String[arraySize][INNER_ARRAY_SIZE];
+            while ((line = bufferedReader.readLine()) != null) {
+                sourceArray[i] = line.split(COMMA);
+                i++;
+            }
+            return sourceArray;
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read info from file", e);
+        }
+    }
+
+    private int[] calculateReport(String[][] originalSourceArray) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+        for (String[] sourceElement : originalSourceArray) {
+            if (sourceElement[0].equals("supply")) {
```

Comment:
supply, buy, result and comma should be constants
0 and 1 are magic number, create constants also